src/orion/core/io/resolve_config.py

```diff
@@ -282,22 +193,66 @@ def merge_orion_config(config, dbconfig, cmdconfig, cmdargs):
     .. seealso:: Method-specific configurations reside in `/config`
 
     """
-    expconfig = deepcopy(config)
-
-    for cfg in (dbconfig, cmdconfig):
-        for k, v in cfg.items():
-            if k in ENV_VARS:
-                for vk, vv in v.items():
-                    expconfig[k][vk] = vv
-            elif v is not None:
-                expconfig[k] = v
-
-    for k, v in cmdargs.items():
-        if v is not None:
-            if k == 'metadata':
-                for vk, vv in v.items():
-                    expconfig[k][vk] = vv
-            else:
-                expconfig[k] = v
-
-    return expconfig
+    return merge_configs(config, dbconfig, cmdconfig, cmdargs)
```

Comment:
Why not just delete this function and replaced the calls to `merge_orion_config` by `merge_configs`?

Reply:
I was planning to do it in the commit adding ExperimentBuilder. I forgot.